editor/src/clj/editor/particle_lib.clj

```diff
@@ -186,6 +186,21 @@
     :type-unsigned-int Graphics$VertexAttribute$DataType/TYPE_UNSIGNED_INT_VALUE
     :type-float Graphics$VertexAttribute$DataType/TYPE_FLOAT_VALUE))
 
+(defn- vector-type->element-count [vector-type]
+  (case vector-type
+    :vector-type-scalar 1
+    :vector-type-vec2 2
+    :vector-type-vec3 3
+    :vector-type-vec4 4
+    :vector-type-mat2 4
+    :vector-type-mat3 9
+    :vector-type-mat4 16))
```

Comment:
This looks exactly like `graphics/attribute-vector-type->component-count`. Maybe rename that to `graphics/vector-type->component-count` and use it instead?